### PR TITLE
Optimize ShortDecimalType to use LongArrayBlock

### PR DESCRIPTION
**Pull request by fmeiser**
Inherit DecimalType from AbstractType instead of AbstractFixedWidthType.
After this change ShortDecimalType can use faster LongArrayBlock.

### CODE REVIEW

**Comment by martint** [comment]
Nice!  Do you have a sense (or measurements) how much this improves performance?

**Comment by fmeiser** [comment]
I have only synthetic results for writing and reading short decimals with a BlockBuilder/Block.
In these tests writing is 6.5 times faster and reading 10 times.

**Comment by fmeiser** [comment]
Travis failed with unrelated error.

File: presto-spi/src/main/java/com/facebook/presto/spi/type/LongDecimalType.java [hunk 1]
```diff
@@ -30,11 +32,38 @@
...
+    public final int getFixedSize()
+    {
+        return SIZE_OF_LONG_DECIMAL;
+    }
+
+    @Override
+    public final BlockBuilder createBlockBuilder(BlockBuilderStatus blockBuilderStatus, int expectedEntries, int expectedBytesPerEntry)
+    {
+        return new FixedWidthBlockBuilder(
+                getFixedSize(),
```
Comment (sopel39) [inline presto-spi/src/main/java/com/facebook/presto/spi/type/LongDecimalType.java]: you can use `SIZE_OF_LONG_DECIMAL` directly here

**Comment by sopel39** [comment]
nice, lgtm

File: presto-spi/src/main/java/com/facebook/presto/spi/type/LongDecimalType.java [hunk 2]
```diff
@@ -30,11 +32,38 @@
 
     LongDecimalType(int precision, int scale)
     {
-        super(precision, scale, Slice.class, SIZE_OF_LONG_DECIMAL);
+        super(precision, scale, Slice.class);
         validatePrecisionScale(precision, scale, MAX_PRECISION);
     }
 
     @Override
+    public final int getFixedSize()
```
Comment (dain) [inline presto-spi/src/main/java/com/facebook/presto/spi/type/LongDecimalType.java]: This class is final, so the methods don't need to be final

**Comment by dain** [comment]
Let us know when the changes are applied.

**Comment by fmeiser** [comment]
@dain Applied your comment and superseded by #5683

**Pull request closed without merging**